oci_tarball: set time, user and group to reproducible values for Linux

gergelyfabian opened the pull request:
Applied suggestion for Linux from:

http://h2.jaguarpaw.co.uk/posts/reproducible-tar/

Left out MacOS and other systems for now (waiting for tar toolchain).
MacOS does not have the same options for tar, for now fixing this only
for Linux.

gergelyfabian commented:
Follow up for #380.

chancila commented:
this is a gnu tar flag, this won't work on other tars...for reference bsd/mac tar uses --uname and --gname

alexeagle commented:
We'll fix this after we have https://github.com/aspect-build/bazel-lib/pull/468 so the same hermetic `tar` is used everywhere.

gergelyfabian commented:
Rebased PR with the revert for #380.
Updated to set parameters only for Linux (I could also set parameters for Mac, but I don't have a way to test it, so preferred to leave the functionality unchanged for Mac). Optimally it would be better to use the tar toolchain (https://github.com/aspect-build/bazel-lib/pull/468), but not having reproducible tars is a regression when migrating from rules_docker to rules_oci, so I'd personally prefer to have it fixed in the meantime.

alexeagle commented:
Sorry, I'd rather not merge a fix that's only for one platform, that just leaves both the original bug and also some tech debt for us to remove. #385 is probably the better change, though it may take a couple weeks to land. I suggest you apply this patch in your own repo for now.

gergelyfabian commented:
Sure. Is #385 already dealing with reproducible tars or it's just starting
to use the tar toolchain?

alexeagle commented:
The `tar` rule in bazel-lib should already be reproducible, that's part of its API contract.

alexeagle commented:
okay I updated that PR with halfway working implementation that uses an mtree file to exactly specify deterministic content of the tar file.

Not sure when @thesayyn or I will get time to finish it...

gergelyfabian commented:
Thank you! Let me know if I can help in any way (I could contribute
something if necessary).

Surely I will test this implementation.

gergelyfabian commented:
Can confirm that #385 already implements reproducible tars.
